storage.c

```diff
@@ -466,31 +466,58 @@ int pv_storage_validate_file_checksum(char *path, char *checksum)
 	return ret;
 }
 
-static bool pv_storage_validate_objects_object_checksum(char *checksum)
+static int pv_storage_get_inode(const char *path, ino_t *inode)
 {
-	char path[PATH_MAX];
+	struct stat st = { 0 };
+
+	if (stat(path, &st) != 0) {
+		pv_log(DEBUG, "couldn't access to file %s, checksum failed",
+		       path);
+		return -1;
+	}
 
-	pv_paths_storage_object(path, PATH_MAX, checksum);
-	pv_log(DEBUG, "validating checksum for object %s", path);
-	return !pv_storage_validate_file_checksum(path, checksum);
+	*inode = st.st_ino;
+	return 0;
 }
 
+static bool pv_storage_compare_files(const char *object, const char *file)
```

Comment:
This function is comparing two generic files, so I'd call it something like is_same_file(path1, path2) and maybe move it to utils/fs.c?